pkg/controller/daemon/daemon_controller.go

```diff
@@ -865,12 +899,29 @@ func (dsc *DaemonSetsController) podsShouldBeOnNode(
 				continue
 			}
 			if pod.Status.Phase == v1.PodFailed {
+				failedPodsObserved++
+
+				// This is a critical place where DS is often fighting with kubelet that rejects pods.
+				// We need to avoid hot looping and backoff.
+				backoffKey := fmt.Sprintf("%s/%s", dsKey, node.Name)
```

Comment:
so this is basically a "namespace/name/node" ?

Reply:
maybe make a function for making a backoff key that takes a ds and a node name to use between this and tests?

Reply:
should DS generation or template generation be part of the backoff key, so that updating the DS spec resets the backoff?